TestGeneratorSystem/src/testgeneratorsystem/Student.java

```diff
@@ -100,6 +113,69 @@ public double getAverageScore() {
         // Implement logic to calculate average score
         return 0.0; // Placeholder value
     }
+    @Override
+    public void removeAccount(){
+        try {
+           listOfStudents.remove(this); 
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+        System.out.println("the account is removed");
+    }
+    @Override
+    public Student login(){
+    Scanner scanner = new Scanner(System.in);
+    System.out.println("Enter username:");
+    String username = scanner.nextLine();
+    System.out.println("Enter password:");
+    String password = scanner.nextLine(); // Authenticate the user
+    Student authenticatedStudent = Student.authenticate(username, password);
+    if (authenticatedStudent != null){
+        System.out.println("Login successful!");
+        return authenticatedStudent;
+    }
+    else
+    { 
+        System.out.println("Login failed. Please check your username and password.");
+        return null;
+    }
+    }
+    public static Student authenticate(String username, String password) {
```

Comment:
needs to be refactored later on to be the same as login function in admin (don't make any current changes)